utxoscanner.go

```diff
@@ -0,0 +1,387 @@
+package neutrino
+
+import (
+	"container/heap"
+	"github.com/davecgh/go-spew/spew"
+	"github.com/roasbeef/btcd/chaincfg/chainhash"
+	"github.com/roasbeef/btcd/wire"
+	"github.com/roasbeef/btcutil"
+	"github.com/roasbeef/btcutil/gcs"
+	"github.com/roasbeef/btcutil/gcs/builder"
+	"github.com/roasbeef/btcwallet/waddrmgr"
+	"sync"
+)
+
+// GetUtxoRequest is a request to scan for OutPoint from the height StartHeight.
+type GetUtxoRequest struct {
+	OutPoint    *wire.OutPoint
+	StartHeight uint32
+	Result      func(*SpendReport, error)
+}
+
+// Interface exposes the necessary methods for interacting with the blockchain.
+type Interface interface {
```

Comment:
While this is easier, the method names look strange since they are tied to the ChainService, making it little different than just adding a `*ChainService` in here..

Did you consider passing methods in a `Config` struct? Sorta like this:
```go
type UtxoScannerConfig struct {
	GetBlock           func(chainhash.Hash, ...QueryOption) (*btcutil.Block, error)
	GetBlockHash       func(int64) (*chainhash.Hash, error)
	BestSnapshot       func() (*waddrmgr.BlockStamp, error)
	GetCFilter         func(blockHash chainhash.Hash, filterType wire.FilterType, options ...QueryOption) (*gcs.Filter, error)
	BlockFilterMatches func(ro *rescanOptions, blockHash *chainhash.Hash) (bool, error)
}
```

Could potentially also make the testing easier..